src/reggae/reggae.d

```diff
@@ -321,30 +352,47 @@ private Binary buildReggaefileDub(O)(auto ref O output, in Options options) {
     // a custom compiler
     return Binary(
         getBuildGenName(options),
-        ["dub", "build"], // since we now depend on dub at buildgen runtime
+        ["dub", "build", "--nodeps", "--skip-registry=all"], // since we now depend on dub at buildgen runtime
     );
 }
 
+private void writeIfDiffers(O)(auto ref O output, in string path, in string contents) @safe {
+    import reggae.io: log;
+    import std.file: exists, readText, write;
+    import std.array: replace;
+
+    if(!path.exists || path.readText.replace("\r\n", "\n") != contents) {
+        output.log("Writing ", path);
+        write(path, contents);
+    }
+}
+
+
 private string[] dubImportFlags(in imported!"reggae.options".Options options) {
-    import std.json: parseJSON;
-    import std.file: exists;
     import reggae.dub.interop.dublib: fetchDubDeps, dubPackagePath;
+    import std.file: exists;
     import std.path: buildPath;
 
     fetchDubDeps(hiddenDirAbsPath(options));
-    const dubSelectionsJson = import("dub.selections.json");
-    const dubVersion = dubSelectionsJson
-        .parseJSON
-        ["versions"]
-        ["dub"]
-        .str;
 
-    const dubSourcePath = buildPath(dubPackagePath("dub", dubVersion), "source");
+    const dubSourcePath = buildPath(dubPackagePath("dub", selectionsPkgVersion("dub")), "source");
     assert(dubSourcePath.exists, "dub fetch failed: no path '" ~ dubSourcePath ~ "'");
 
     return ["-I" ~ dubSourcePath];
 }
 
+// the unit-threaded version we depend on
+private string selectionsPkgVersion(in string pkg) @safe pure {
+    import std.json: parseJSON;
+
+    return import("dub.selections.json")
+        .parseJSON
+        ["versions"]
+        [pkg]
+        .str;
+}
```

Comment:
I'm not sure how I could. What I'm doing here is... weird. I'm writing a dub.selections.json file for a dub project that "doesn't exist": it's a dummy dub package used to compile the build description itself. I want to make sure it depends on the same version of dub that reggae itself depends on, and because of the way configurations are merged, the same version of unit-threaded even though it's not actually being used.

Reply:
So you are relying on the very bug I'm trying to fix ? 🙃 
https://xkcd.com/1172/

Reply:
*Nearly*. The only part I'm depending on right now is unit-threaded, since that shouldn't be here.